enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,789 @@
+
+## Summary
+
+There's an ongoing interest to migrate Virtual Machines (VMs) from other
+virtualization platforms into OpenShift. These users are both after a managed IP
+experience (which they have on their existing platform), and also want to
+consume existing Kubernetes features, like network policies, and sometimes
+services.
+
+Traditional virtualization users have some expectations on what a
+virtualization platform should provide. Live-migration, and IP address
+persistence across reboots are paramount features for any virtualization
+solution, and OpenShift Virtualization currently supports these features on its
+primary UserDefinedNetworks (UDNs).
+
+These users have additional requirements, like routed ingress into their VMs,
+and, to import the VMs with the MAC, IPs, and gateway configuration the VM had
+on the previous platform.
+
+Routed ingress into the VMs is being added to OpenShift already, and is tracked
+by [this enhancement](https://github.com/openshift/enhancements/pull/1636/files)
+which tracks adding BGP support to OpenShift.
+
+Given the aggressive time-frame we're attempting, adding support for importing
+a VM into OpenShift with it's MAC, IPs, and gateway on top of primary UDN is the
+missing piece we should pursue.
+
+## Motivation
+
+Some users are running VMs in virtualization platforms having a managed IP
+configuration.
+
+For this kind of users, we need to have a way to enable their existing VMs to
+run properly after being migrated into OpenShift, without any guest
+configuration changes. For that, we need to import the VMs from their existing
+platforms, preserving their existing MACs, IPs, and gateway configuration.
+
+### User Stories
+
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM - whose IPs were statically configured - into Kubernetes,
+attaching it to an overlay network. I want to ingress/egress using the same IP
+address (the one assigned to the VM).
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM into Kubernetes, attaching it to an overlay network. I want to
+be able to consume Kubernetes features like network policies, and services, to
+benefit from the Kubernetes experience.
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM into Kubernetes, attaching it to an overlay network, without
+having to reconfigure the VM's networking configuration (MAC, IPs, gateway).
+
+### Goals
+
+- Preserve the original VM's MAC address
+- Preserve the original VM's IP address
+- Specify the gateway IP address of the imported VM so it can keep the same
+default route. This gateway is common to the entire logical network.
+- Ensure it is possible to enable non-NATed traffic for pods with the static
+network configuration by exposing the network through BGP
+
+**NOTE:** all the goals mentioned above will be fulfilled **only** for the
```

Comment:
Most of the goals will work for non-cluster UDNs. Can we just highlight that today BGP only works with cluster UDNs?